Orange/preprocess/fss.py

```diff
@@ -50,6 +51,12 @@ def __init__(self, method=None, k=None, threshold=None, decreasing=True):
         self.decreasing = decreasing
 
     def __call__(self, data):
+        n_attrs = len(data.domain.attributes)
+        if isinstance(self.k, float):
+            idx_attr = np.ceil(self.k * n_attrs).astype(int)
+            # edge case: 0th percentile would result in selection of `(n_attrs + 1)` attrs
+            self.k = min(n_attrs - idx_attr + 1, n_attrs)
```

Comment:
`__call__` shouldn't change the state of the object (unless when caching, for instance). In this case, imagine you have

```
fss = SelectBestFeatures(k=0.5)
data1 = <some data set with 10 attributes>
data2 = <some data set with 100 attributes>
fss(data1)  # this sets fss.k to 5!
fss(data2)  # and so this selects just 5 instead of 50 attributes
```

One option is to introduce

```
            effective_k = min(n_attrs - idx_attr + 1, n_attrs)
    else:
            effective_k = self.k
```

and replace further occurrences of `self.k` in this method with `effective_k`.

Reply:
Good catch, I didn't think about that.  
  
I've fixed it (done the same thing for selection of random features as well, since it had the same problem) and added a test for this.